addons/bestja_detailed_reports/models.py

```diff
@@ -0,0 +1,215 @@
+        store=True,
+    )
+    user_can_moderate = fields.Boolean(compute="compute_user_can_moderate")
+
+    @api.one
+    @api.depends('parent_project', 'project')
+    def compute_interested_organization(self):
+        level = self.project.organization.level
+        if level <= 1:
+            self.interested_organization = self.project.organization.id
+        else:
+            self.interested_organization = self.parent_project.organization.id
+
+    @api.one
+    @api.depends('report_entries')
+    def compute_report_tonnage(self):
+        """
+        For showing the sum of kg of products
+        """
+        self.tonnage = 0.0
+        for q in self.report_entries:
+            self.tonnage += q.tonnage
+
+    @api.one
+    @api.depends('parent_project')
+    def compute_user_can_moderate(self):
+        """
+        Is current user authorized to moderate (accept/reject) the detailed_report?
+        """
+        self.user_can_moderate = (
+            self.parent_project.manager.id == self.env.user.id or
+            self.parent_project.organization.coordinator.id == self.env.user.id
+        )
+
+    @api.one
+    @api.constrains('report_entries')
+    def check_report_entries(self):
+        listlen = len(self.report_entries)
+        newlist = set((i.commodity_code, i.detailed_report) for i in self.report_entries)
+        if (listlen > len(newlist)):
+            raise exceptions.ValidationError("Produkt o danym kodzie może mieć co najwyżej jeden wpis!")
+
+    @api.one
+    @api.depends('project')
+    def compute_project_dates(self):
+        """
+        For nice format in the list of all projects.
+        """
+        if self.project:
+            tmp_date = parser.parse(self.project.date_start).strftime("%d-%m-%Y")
+            tmp_date += " -- "
+            tmp_date += parser.parse(self.project.date_stop).strftime("%d-%m-%Y")
+            self.dates = tmp_date
+
+    @api.multi
+    def set_sent(self):
+        if (self.project.organization.level <= 1):
+            self.state = 'accepted'
+        else:
+            self.state = 'sent'
+            self.send(
+                template='bestja_detailed_reports.msg_detailed_report_sent',
+                recipients=self.parent_project.responsible_user,
+                record_name=self.organization.name,
+            )
+
+    @api.multi
+    def set_accepted(self):
+        self.state = 'accepted'
+
+    @api.multi
+    def set_rejected(self):
+        self.state = 'rejected'
+        self.send(
+            template='bestja_detailed_reports.msg_detailed_reports_rejected',
+            recipients=self.sudo().project.responsible_user,
+            sender=self.env.user,
+        )
+
+    @api.multi
+    def continue_action(self):
+        """
+        For the continue button.
+        """
+        pass
+
+    @api.multi
+    def add_all_commodity_groups(self):
+        """
+        Button which adds all commodity groups to the report
+        """
+        codes = []
```

Comment:
What do you use this list for? I see that (despite its name) you add  `commodity_group` objects to it, but I don't see you actually using it anywhere.